qemu/tests/virtio_fs_supp_group_transfer.py

```diff
@@ -0,0 +1,84 @@
+import os.path
+
+from virttest import error_context, utils_disk, utils_misc
+
+from provider import virtio_fs_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Grant write access at users allowed by group permission
+    on shared dir( linux only )
+
+    1) Run the virtiofsd daemon on the host and Set log level to "debug".
+    2) Boot a guest on the host.
+    3) Log into guest with root, then mount the file system.
+    4) In the guest, add a new user(u1)
+       and set the user to the wheel group (as supplementary group)
+       and set the dir's group to 'wheel'
+    5) In guest, using root to create a test directory at the shared dir.
+    6) Run the basic io test with the u1 user
+    7) After test, clear the environment.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    username = params.get("new_guest_user")
+    fs_dest = params.get("fs_dest")
+    fs_target = params.get("fs_target")
+
+    testdir = "testdir"
+    guest_root_session = None
+    guest_user_session = None
+
+    try:
+        shared_dir = params.get("fs_source_dir")
+
+        vm = env.get_vm(params.get("main_vm"))
+        vm.verify_alive()
+        guest_root_session = vm.wait_for_login()
+
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, test.log.info)
+        if not utils_misc.make_dirs(fs_dest, session=guest_root_session):
+            test.fail("Creating directory was failed!")
+        error_context.context("Mount virtiofs target %s to %s inside"
+                              " guest." % (fs_target, fs_dest),
+                              test.log.info)
+        if not utils_disk.mount(fs_target, fs_dest, 'virtiofs',
+                                session=guest_root_session):
+            test.fail('Mount virtiofs target failed.')
+
+        guest_root_session.cmd(add_user_cmd % username)
+        guest_root_session.cmd("usermod -G wheel %s" % username)
+
+        guest_root_session.cmd("cd %s && mkdir -m 770 %s" % (fs_dest, testdir))
+        guest_root_session.cmd("cd %s && chgrp wheel %s" % (fs_dest, testdir))
+        params["fs_dest"] = os.path.join(fs_dest, testdir)
+        params["fs_source_dir"] = os.path.join(shared_dir, testdir)
```

Comment:
Note that here we modified fs_source_dir, and in the end, VT will remove it, ie. VT removes /home/kar/workspace/root/avocado/data/avocado-vt/virtio_fs_test/testdir instead of /home/kar/workspace/root/avocado/data/avocado-vt/virtio_fs_test(created by VT), so finally, /home/kar/workspace/root/avocado/data/avocado-vt/virtio_fs_test  still exists, we need to handle this

Reply:
We may make a copy of params, with fs_source_dir changed, then pass that copy to basic_io_test